mesonbuild/compilers/compilers.py

```diff
@@ -763,6 +763,8 @@ def build_unix_rpath_args(self, build_dir, from_dir, rpath_paths, install_rpath)
         # directory. This breaks reproducible builds.
         rel_rpaths = []
         for p in rpath_paths:
+            if not p:
```

Comment:
This should probably be `p == ''` to be perfectly clear what the case is. Please also add a comment here explaining why this condition exists.